### DISCUSSION TITLE
Loading.tsx conflicting with nested layout.

### DISCUSSION THREAD

**Discussion by marif0025** [vercel/next.js#53383] [App Router] [answered]
Hi,

Recently, I implemented a loading feature in my application, wherein I have a loading.tsx component placed at the application's root level, and there are other loading components nested within specific routes.

The issue I am encountering is that when I navigate to a nested route, both the root loading screen and the loading screen of the particular route are displayed consecutively. Consequently, the content of the nested route is only visible after both loading screens have been shown.

I am seeking your expertise in resolving this matter. Could you please guide me on the appropriate steps to rectify this situation and ensure that only the relevant loading screen appears when accessing a nested route, and the content is displayed without undue delay?

Thank you in advance for your assistance, and I look forward to your valuable input.

**Comment by Nadeera3784** [comment]
1. Avoid displaying the root loading component on nested routes:

In your root-level loading component (e.g., `loading.tsx`), you can use React Router or Next.js built-in hooks to determine whether you are on a nested route or not. If you are on a nested route, you can conditionally choose not to render the root-level loading component. This way, only the relevant loading component for the specific nested route will be displayed.

Here's an example of how you can conditionally render the root loading component only for non-nested routes:

```jsx
// Root-level loading component (loading.tsx)
import { useRouter } from 'next/router';

const RootLoading = () => {
  const router = useRouter();
  
  // Check if the current route is a nested route
  const isNestedRoute = router.pathname.split('/').length > 2;
  
  // Render the loading component only for non-nested routes
  if (isNestedRoute) {
    return null;
  }
  
  return (
    // Your root-level loading screen JSX here
  );
};

export default RootLoading;
```

2. Implement loading components for specific nested routes:

For each specific nested route that you want to display a loading screen, create a separate loading component and render it in the respective nested route component. This way, when you navigate to a nested route, only the relevant loading component for that specific route will be displayed, and the root-level loading component won't interfere.

Here's an example of how to create a loading component for a nested route:

```jsx
// Nested route loading component (nestedLoading.tsx)
const NestedLoading = () => {
  return (
    // Your nested route loading screen JSX here
  );
};

export default NestedLoading;
```

3. Use the nested loading component in the specific nested route:

In your specific nested route component (e.g., `pages/nestedRoute/[id].tsx`), import and render the nested loading component at the appropriate place. This loading component will only be displayed when users access that particular nested route.

```jsx
// Specific nested route component (pages/nestedRoute/[id].tsx)
import NestedLoading from './path/to/nestedLoading';

const NestedRoute = () => {
  // Your nested route component logic here
  
  return (
    <>
      {/* Render the nested loading component */}
      <NestedLoading />
      
      {/* Your nested route content JSX here */}
    </>
  );
};

export default NestedRoute;
```

By following these steps, you should be able to show only the relevant loading component for each specific nested route, and the content should be displayed without undue delay. The root-level loading component will only be displayed for non-nested routes.

**Reply by marif0025** [reply, marked as answer]
Option 1 _(Avoid displaying the root loading component on nested routes:)_ works for me.
Just want to add here that you have to use `usePathname() `hook to get pathname and this component will be client component.

**Comment by fnb-software** [comment]
It's weird to me that loading screens would be nested. 
Also why would a loading screen be needed if the route navigated to isn't an `async` component?
